hack/make/.detect-daemon-osarch

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
```

Comment:
This is also from @tianon's PR #19076 as we need these variables for `build-deb` and `build-rpm` scripts to select the correct osarch. We will rebase after 19076 get merged ;-)